### PR TITLE
Add support for command line config parsing

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/28894

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by sameerajayasoma** [comment]
I noticed that some Diagnostic related classes are duplicated in this PR. Can we simply take a dependency on the `ballerina-tools-api` library?

**Comment by warunalakshitha** [comment]
> I noticed that some Diagnostic related classes are duplicated in this PR. Can we simply take a dependency on the `ballerina-tools-api` library?

@sameerajayasoma Yes. I am working on separate PR for use Diagnostic from tools api which required minor refactoring to toml related classes too with configurable variable location, proper message format etc.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/ConfigResolver.java [hunk 1]
```diff
@@ -46,100 +48,137 @@
...
+                    configValue = value;
+                }
+            } catch (ConfigException e) {
+                exceptionList.add(e);
+            }
+        }
+
+        // Handle errors while getting config values.
+        if (configValue.isPresent()) {
+            exceptionList.forEach(e -> diagnosticLog.warn(e.getMessage()));
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/ConfigResolver.java:166]: This is the logic we check whether we print war or error. If at least one providers has a value we print them as warnings.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/cli/CliProvider.java [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+        } catch (BError e) {
+            throw new CliConfigException(cliArg, String.format(INCOMPATIBLE_TYPE_ERROR_MESSAGE, key.variable,
+                                                               effectiveType, cliArg.value));
+        }
+    }
+
+    private CliArg getCliArg(Module module, VariableKey variableKey) {
+        String key = variableKey.variable;
+        String value;
+        if (rootModule.getOrg().equals(module.getOrg())) {
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/cli/CliProvider.java:215]: do we consider the `DEFAULT_MODULE` case? 
For single file execution, how can the user provide args?

Comment (warunalakshitha) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/cli/CliProvider.java:215]: sinfle bal file case rootModule = module. So this logic will work for single bal files too.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlFileProvider.java [hunk 3]
```diff
@@ -38,21 +37,22 @@
...
 
     @Override
-    public void initialize(Map<Module, VariableKey[]> configVarMap) {
+    public void initialize() {
         super.tomlNode = getConfigTomlData(configPath);
-        super.initialize(configVarMap);
     }
 
     private TomlTableNode getConfigTomlData(Path configFilePath) {
+        if (!Files.exists(configFilePath)) {
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlFileProvider.java:49]: We can remove this check as we already checking this at `LaunchUtils`.

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java [hunk 4]
```diff
@@ -220,39 +221,40 @@ private Object getPrimitiveTomlValue(Module module, VariableKey key) {
...
 
-    private Set<String> getRequiredModules(Map<Module, VariableKey[]> configVarMap) {
-        Set<String> modules = new HashSet<>();
-        for (Map.Entry<Module, VariableKey[]> moduleEntry : configVarMap.entrySet()) {
-            for (VariableKey key : moduleEntry.getValue()) {
-                if (key.isRequired()) {
-                    modules.add(key.module.getName());
-                }
-            }
+    private TomlTableNode getConfigurationData(Path configFilePath) {
```
Comment (HindujaB) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/TomlProvider.java:244]: This logic is already there in `TomlFileProvider`. Shall we remove this?s

**Review by manuranga — APPROVED** [review]

**Pull request merged**